### DISCUSSION TITLE
Could I use a base64 content of the .mid file in src?

### DISCUSSION THREAD

**Discussion by elloza** [cifkao/html-midi-player#77] [Q&A] [answered]
Could I use a base64 content of the .mid file in src?

I'm trying to load a .mid in Streamlit with this component but I can't include the path to the file, could I embed it in base64 somehow to make it work?

Do you have any ideas?

Thanks a lot!

**Comment by cifkao** [comment, marked as answer]
Yes, I think this works! It's what @drscotthawley has done [here](https://github.com/drscotthawley/midi-player/blob/c4929a7614d9d8b050cde1ac0a731aff32a3b51d/midi_player/midi_player.py#L50-L52) in his `midi-player` Python package.

**Reply by elloza** [reply]
Yes, I finally got it working! Thank you so much!

**Reply by elloza** [reply]
```
    mp = MIDIPlayer(midi_path, 300, viz_type="waterfall")
    # HTML para el reproductor MIDI
    html_content = f"""
    <html>
    <body>
    {mp.html}
    <script src="https://cdn.jsdelivr.net/combine/npm/tone@14.7.58,npm/@magenta/music@1.23.1/es6/core.js,npm/focus-visible@5,npm/html-midi-player@1.5.0"></script>
    </body>
    </html>
    """
    st.components.v1.html(html_content, width=700, height=300)

```

Maybe a little bit complicated, but it's working.
